### PR TITLE
Feature mes 2310 fault visibility

### PR DESCRIPTION
**Pull request by rossfauldsbjss**
MES 2310 - Manoeuvre, Vehicle check, and Controlled stop faults (driving, dangerous, and serious) were not visible on bottom of write up page - or available for entering comments into if applicable. 
Modified  to ensure that they displayed, were editable if applicable, and persisted.

This work was done - but there may be a future PR to refactor comment/ fault locations in the schema to enable cleaner code to be written. 
## Pull Request checklist:

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop
- [ ] commits are squashed

## Sign off process checklist:

- [x] Code has been tested manually
- [x] Tests are passing
- [x] PR link added to JIRA
- [x] Reviewers selected in Github
- [ ] Any new reusable component/widget added to component library on Confluence

- [ ] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [ ] PO's approval
IMG_713BD33E1B2C-1
IMG_713BD33E1B2C-2

### CODE REVIEW

File: src/modules/tests/test-data/test-data.actions.ts [hunk 1]
```diff
@@ -33,11 +35,13 @@ export const CONTROLLED_STOP_ADD_DRIVING_FAULT = '[ControlledStop] Add Driving F
 export const CONTROLLED_STOP_ADD_SERIOUS_FAULT = '[ControlledStop] Add Serious Fault';
 export const CONTROLLED_STOP_ADD_DANGEROUS_FAULT = '[ControlledStop] Add Dangerous Fault';
 export const CONTROLLED_STOP_REMOVE_FAULT = '[ControlledStop] Remove Fault';
+export const CONTROLLED_STOP_ADD_COMMENT = '[ControlledStop] Add Comment';
```
Comment (johnmorleybjss) [inline src/modules/tests/test-data/test-data.actions.ts]: The name here isn't quite consistent with the other new actions you added. `ADD_MANOEUVRE_COMMENT` vs `CONTROLLED_STOP_ADD_COMMENT`.

File: src/modules/tests/test-data/test-data.actions.ts [hunk 2]
```diff
@@ -71,6 +75,15 @@ export class AddManoeuvreDangerousFault implements Action {
   readonly type = ADD_MANOEUVRE_DANGEROUS_FAULT;
 }
 
+export class AddManoeuvreComment implements Action {
+  constructor(
+    public fieldName: string,
+    public faultType: CompetencyOutcome,
+    public controlOrObservation: string,  // 'Control' | 'Observation',
```
Comment (johnmorleybjss) [inline src/modules/tests/test-data/test-data.actions.ts]: This feels like something we should be modelling in test schema.

File: src/modules/tests/test-data/test-data.selector.ts [hunk 3]
```diff
@@ -72,6 +72,9 @@ export const getDangerousFaultSummaryCount = (data: TestData): number => {
 
 export const sumVehicleCheckFaults = (vehicleChecks: VehicleChecks): number => {
 
+  if (!vehicleChecks || !vehicleChecks.showMeQuestion || !vehicleChecks.tellMeQuestion) {
```
Comment (johnmorleybjss) [inline src/modules/tests/test-data/test-data.selector.ts]: Is this guarding against a newly introduced NPE, or was it a speculative fix? Do we still need it?

File: src/pages/debrief/debrief.selector.ts [hunk 4]
```diff
@@ -71,106 +62,244 @@ export const getManoeuvreFaults = (
...
+          faultCount = faultCount + 1;
+        }
+      });
+    }
+  });
+  return faultCount;
+};
 
-  vehicleChecks.showMeQuestion.outcome === CompetencyOutcome.D && result.push('vehicleChecks');
+export const getVehicleCheckDangerousFault =
```
Comment (johnmorleybjss) [inline src/pages/debrief/debrief.selector.ts]: Singular function name, but returning an array.

File: src/modules/tests/test-data/test-data.actions.ts (hunk 1 above)
Comment (rossfauldsbjss) [reply src/modules/tests/test-data/test-data.actions.ts]: Will rename to be consistent

File: src/modules/tests/test-data/test-data.selector.ts (hunk 3 above)
Comment (rossfauldsbjss) [reply src/modules/tests/test-data/test-data.selector.ts]: it was created to guard against an issue during dev - however upon dev completion I don't think the situation is happening so can remove this guard clause.

File: src/pages/debrief/debrief.selector.ts (hunk 4 above)
Comment (rossfauldsbjss) [reply src/pages/debrief/debrief.selector.ts]: pluralising.

**Pull request merged**